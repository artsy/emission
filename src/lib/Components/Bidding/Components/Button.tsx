import React from "react"
import { StyleSheet } from "react-native"

import { ButtonProps, InvertedButton } from "../../Buttons"

export class Button extends React.Component<ButtonProps> {
  render() {
    const { style, ...props } = this.props

    return <InvertedButton style={[styles.default, style]} {...props} />
  }
}

const styles = StyleSheet.create({
  default: {
    // fontSize: 14,
    // fontWeight: 'normal',
    // fontStyle: 'normal',
    // letterSpacing: 1,
    // color: '#ffffff'
  },
})
