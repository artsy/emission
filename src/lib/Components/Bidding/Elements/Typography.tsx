import React from "react"
import styled from "styled-components/native"
import { color, fontSize, lineHeight, maxWidth, space, textAlign } from "styled-system"

import { Fonts } from "../../../data/fonts"

const Sans = styled.Text`
  font-family: "${Fonts.Unica77LLRegular}";
  ${color}
  ${fontSize}
  ${lineHeight}
  ${space}
  ${textAlign}
  ${maxWidth}
`

const Serif = styled.Text`
  font-family: "${Fonts.GaramondRegular}";
  ${color}
  ${fontSize}
  ${lineHeight}
  ${space}
  ${textAlign}
  ${maxWidth}
`

const SerifSemibold = styled.Text`
  font-family: "${Fonts.GaramondSemibold}";
  ${color}
  ${fontSize}
  ${lineHeight}
  ${space}
  ${textAlign}
  ${maxWidth}
`

const SerifItalic = styled.Text`
  font-family: "${Fonts.GaramondItalic}";
  ${color}
  ${fontSize}
  ${lineHeight}
  ${space}
  ${textAlign}
  ${maxWidth}
`

export const Sans12 = props => <Sans fontSize={1} lineHeight={1} {...props} />
export const Sans14 = props => <Sans fontSize={2} lineHeight={5} {...props} />
export const Sans16 = props => <Sans fontSize={3} lineHeight={6} {...props} />
export const Sans18 = props => <Sans fontSize={4} lineHeight={8} {...props} />

export const Serif12 = props => <Serif fontSize={1} lineHeight={1} {...props} />
export const Serif14 = props => <Serif fontSize={2} lineHeight={2} {...props} />
export const Serif16 = props => <Serif fontSize={3} lineHeight={5} {...props} />
export const Serif18 = props => <Serif fontSize={4} lineHeight={6} {...props} />

export const SerifSemibold12 = props => <SerifSemibold fontSize={1} lineHeight={1} {...props} />
export const SerifSemibold14 = props => <SerifSemibold fontSize={2} lineHeight={2} {...props} />
export const SerifSemibold16 = props => <SerifSemibold fontSize={3} lineHeight={5} {...props} />
export const SerifSemibold18 = props => <SerifSemibold fontSize={4} lineHeight={6} {...props} />
export const SerifSemibold22 = props => <SerifSemibold fontSize={5} lineHeight={9} {...props} />

export const SerifItalic12 = props => <SerifItalic fontSize={1} lineHeight={1} {...props} />
export const SerifItalic14 = props => <SerifItalic fontSize={2} lineHeight={2} {...props} />
export const SerifItalic16 = props => <SerifItalic fontSize={3} lineHeight={5} {...props} />
export const SerifItalic18 = props => <SerifItalic fontSize={4} lineHeight={6} {...props} />
