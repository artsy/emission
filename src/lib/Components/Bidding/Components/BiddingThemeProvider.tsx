import React from "react"
import { ThemeProvider } from "styled-components"
import { theme } from "../Elements/Theme"

export class BiddingThemeProvider extends React.Component {
  render() {
    return <ThemeProvider theme={theme} {...this.props} />
  }
}
