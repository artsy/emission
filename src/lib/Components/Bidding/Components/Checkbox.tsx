import React, { Component } from "react"
import { StyleSheet, TouchableWithoutFeedback, TouchableWithoutFeedbackProperties } from "react-native"
import styled from "styled-components/native"

import { Flex, FlexProps } from "../Elements/Flex"
import { theme } from "../Elements/Theme"
import { CssTransition } from "./Animation/CssTransition"

interface CheckboxState {
  checked: boolean
}

interface CheckboxProps extends TouchableWithoutFeedbackProperties, FlexProps {
  checked?: boolean
  disabled?: boolean
  error?: boolean
}

export class Checkbox extends Component<CheckboxProps, CheckboxState> {
  private readonly checkboxSize = 20

  private readonly duration = 250

  private readonly defaultCheckboxStyle = {
    backgroundColor: theme.colors.white100,
    borderColor: theme.colors.black10,
  }

  private readonly checkedCheckboxStyle = {
    backgroundColor: theme.colors.black100,
    borderColor: theme.colors.black100,
  }

  private readonly disabledCheckboxStyle = {
    backgroundColor: theme.colors.black5,
    borderColor: theme.colors.black10,
  }

  private readonly checkboxStyles = {
    default: {
      unchecked: this.defaultCheckboxStyle,
      checked: this.checkedCheckboxStyle,
    },
    error: {
      unchecked: Object.assign({}, this.defaultCheckboxStyle, { borderColor: theme.colors.red100 }),
      checked: this.checkedCheckboxStyle,
    },
  }

  constructor(props) {
    super(props)

    this.state = {
      checked: props.checked || false,
    }
  }

  onPress = event => {
    if (this.props.onPress) {
      this.props.onPress(event)
    }

    this.setState({
      checked: !this.state.checked,
    })
  }

  render() {
    const { error, disabled, children, ...props } = this.props
    const { checked } = this.state

    let checkboxStyle
    if (disabled) {
      checkboxStyle = this.disabledCheckboxStyle
    } else {
      checkboxStyle = this.checkboxStyles[error ? "error" : "default"][checked ? "checked" : "unchecked"]
    }

    return (
      <TouchableWithoutFeedback onPress={disabled ? () => null : this.onPress}>
        <Flex alignItems="center" flexDirection="row" {...props}>
          <CssTransition
            style={[styles.container, checkboxStyle]}
            animate={["backgroundColor", "borderColor"]}
            duration={this.duration}
          >
            {checked && (disabled ? <DisabledMark size={this.checkboxSize} /> : <CheckMark size={this.checkboxSize} />)}
          </CssTransition>

          <Flex>{children}</Flex>
        </Flex>
      </TouchableWithoutFeedback>
    )
  }
}

// styled-component does not have support for Animated.View
const styles = StyleSheet.create({
  container: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    borderWidth: 2,
    borderStyle: "solid",
    marginRight: theme.space[3],
    width: 20,
    height: 20,
  },
})

interface CheckMarkProps {
  size: number
}

// This component represents the âˆš mark in CSS. We are not using styled-system since it's easier to specify raw CSS
// properties with styled-component.
export const CheckMark = styled.View.attrs<CheckMarkProps>({})`
  transform: rotate(-45deg);
  top: -12%;
  width: ${props => props.size * 0.625};
  height: ${props => props.size * 0.3125};
  border-bottom-color: white;
  border-bottom-width: 2px;
  border-left-color: white;
  border-left-width: 2px;
`

export const DisabledMark = styled(CheckMark)`
  border-bottom-color: ${theme.colors.black30};
  border-left-color: ${theme.colors.black30};
`
