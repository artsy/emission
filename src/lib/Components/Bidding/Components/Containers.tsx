import React from "react"
import { Flex } from "../Elements/Flex"

export const Container = props => (
  <Flex m={4} flex={1} flexDirection="column" justifyContent="space-between" {...props} />
)

export const CenteringContainer = props => <Flex flex={1} justifyContent="center" alignItems="center" {...props} />
