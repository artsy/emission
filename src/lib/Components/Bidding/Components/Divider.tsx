import React from "react"
import { Flex } from "../Elements/Flex"

export const Divider = props => <Flex width="100%" border={1} borderColor="black10" borderBottomWidth={0} {...props} />
