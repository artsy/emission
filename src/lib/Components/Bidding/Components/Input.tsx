import React, { Component } from "react"
import { TextInputProperties } from "react-native"
import { TextInput, TextInputProps } from "../Elements/TextInput"

interface InputProps extends TextInputProps, TextInputProperties {
  error?: boolean
  inputRef?: (component: any) => void
}

class InputState {
  borderColor: "black10" | "purple100" | "red100"
}

export class Input extends Component<InputProps, InputState> {
  constructor(props) {
    super(props)

    this.state = {
      borderColor: this.props.error ? "red100" : "black10",
    }
  }

  componentDidUpdate(prevProps) {
    if (this.props.error !== prevProps.error) {
      this.setState({ borderColor: this.props.error ? "red100" : "black10" })
    }
  }

  onBlur(e) {
    if (this.props.onBlur) {
      this.props.onBlur(e)
    }

    this.setState({
      borderColor: this.props.error ? "red100" : "black10",
    })
  }

  onFocus(e) {
    if (this.props.onFocus) {
      this.props.onFocus(e)
    }

    this.setState({ borderColor: "purple100" })
  }

  render() {
    return (
      <TextInput
        border={1}
        borderColor={this.state.borderColor}
        fontSize={3}
        p={3}
        pb={2}
        {...this.props}
        // These props should not be overridden so they are declared after `{...this.props}`
        ref={this.props.inputRef}
        onBlur={e => this.onBlur(e)}
        onFocus={e => this.onFocus(e)}
      />
    )
  }
}
