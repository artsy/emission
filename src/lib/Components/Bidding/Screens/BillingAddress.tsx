import React from "react"
import { ScrollView } from "react-native"

import { Flex } from "../Elements/Flex"
import { Sans12, Serif16 } from "../Elements/Typography"

import { BiddingThemeProvider } from "../Components/BiddingThemeProvider"
import { Button } from "../Components/Button"
import { Container } from "../Components/Containers"
import { Input } from "../Components/Input"
import { Title } from "../Components/Title"

import { InputStrings, isMinLength, isRequired, Validation } from "./Validation"

export interface Address extends InputStrings {
  fullName: string
  addressLine1: string
  addressLine2?: string
  city: string
  state: string
  postalCode: string
}

interface BillingAddressProps {
  onSubmit?: (values: Address) => null
  billingAddress?: Address
}

interface BillingAddressState {
  values: Address
  errors: {
    fullName?: string
    addressLine1?: string
    addressLine2?: string
    city?: string
    state?: string
    postalCode?: string
  }
}

export class BillingAddress extends React.Component<BillingAddressProps, BillingAddressState> {
  private readonly validationSchema = new Validation({
    fullName: [isRequired()],
    addressLine1: [isRequired()],
    city: [isRequired()],
    state: [isRequired()],
    postalCode: [isRequired(), isMinLength(3, "postal codes are probably at least 3 long")],
  })

  constructor(props) {
    super(props)

    this.state = {
      values: {
        fullName: "",
        addressLine1: "",
        addressLine2: "",
        city: "",
        state: "",
        postalCode: "",
        ...this.props.billingAddress,
      },
      errors: {},
    }
  }

  onSubmit() {
    const errors = this.validationSchema.validate(this.state.values)
    if (Object.keys(errors).length === 0) {
      this.props.onSubmit(this.state.values)
    } else {
      this.setState({ errors })
    }
  }

  render() {
    const { errors } = this.state

    return (
      <BiddingThemeProvider>
        <ScrollView>
          <Container>
            <Title mt={0} mb={6}>
              Your billing address
            </Title>

            <Serif16 mb={2}>Full name</Serif16>
            <StyledInput
              value={this.state.values.fullName}
              onChangeText={fullName => this.setState({ values: { ...this.state.values, fullName } })}
              placeholder="Enter your full name"
              error={errors.fullName}
            />

            <Serif16 mb={2}>Address line 1</Serif16>
            <StyledInput
              value={this.state.values.addressLine1}
              onChangeText={addressLine1 => this.setState({ values: { ...this.state.values, addressLine1 } })}
              placeholder="Enter your street address"
              error={errors.addressLine1}
            />

            <Serif16 mb={2}>Address line 2 (optional)</Serif16>
            <StyledInput
              value={this.state.values.addressLine2}
              onChangeText={addressLine2 => this.setState({ values: { ...this.state.values, addressLine2 } })}
              placeholder="Enter your apt, floor, suite, etc."
              error={errors.addressLine2}
            />

            <Serif16 mb={2}>City</Serif16>
            <StyledInput
              value={this.state.values.city}
              onChangeText={city => this.setState({ values: { ...this.state.values, city } })}
              placeholder="Enter city"
              error={errors.city}
            />

            <Serif16 mb={2}>State, Province, or Region</Serif16>
            <StyledInput
              value={this.state.values.state}
              onChangeText={state => this.setState({ values: { ...this.state.values, state } })}
              placeholder="Enter state, province, or region"
              error={errors.state}
            />

            <Serif16 mb={2}>Postal code</Serif16>
            <StyledInput
              value={this.state.values.postalCode}
              onChangeText={postalCode => this.setState({ values: { ...this.state.values, postalCode } })}
              placeholder="Enter your postal code"
              error={errors.postalCode}
            />

            <Button text="Add billing address" onPress={() => this.onSubmit()} />
          </Container>
        </ScrollView>
      </BiddingThemeProvider>
    )
  }
}

const StyledInput = ({ error, ...props }) => (
  <Flex mb={4}>
    <Input mb={3} error={Boolean(error)} {...props} />
    {error && <Sans12 color="red100">{error}</Sans12>}
  </Flex>
)
