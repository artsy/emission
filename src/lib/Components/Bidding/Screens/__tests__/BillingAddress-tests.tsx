import React from "react"
import { TextInput } from "react-native"
import * as renderer from "react-test-renderer"

import { Button } from "../../Components/Button"
import { Sans12 } from "../../Elements/Typography"
import { BillingAddress } from "../BillingAddress"

it("renders properly", () => {
  const component = renderer.create(<BillingAddress />).toJSON()
  expect(component).toMatchSnapshot()
})

it("shows an error mesasge for each field", () => {
  const component = renderer.create(<BillingAddress />)

  component.root.findByType(Button).instance.props.onPress()

  expect(errorTextComponent(component, "Full name").props.children).toEqual("Required")
  expect(errorTextComponent(component, "Address line 1").props.children).toEqual("Required")
  expect(errorTextComponent(component, "City").props.children).toEqual("Required")
  expect(errorTextComponent(component, "State, Province, or Region").props.children).toEqual("Required")
  expect(errorTextComponent(component, "Postal code").props.children).toEqual("Required")
})

it("calls the onSubmit() callback with billing address when ADD BILLING ADDRESS is tapped", () => {
  const onSubmitMock = jest.fn()
  const component = renderer.create(<BillingAddress onSubmit={onSubmitMock} />)

  textInputComponent(component, "Full name").props.onChangeText("Yuki Stockmeier")
  textInputComponent(component, "Address line 1").props.onChangeText("401 Broadway")
  textInputComponent(component, "Address line 2 (optional)").props.onChangeText("25th floor")
  textInputComponent(component, "City").props.onChangeText("New York")
  textInputComponent(component, "State, Province, or Region").props.onChangeText("NY")
  textInputComponent(component, "Postal code").props.onChangeText("10013")
  component.root.findByType(Button).instance.props.onPress()

  expect(onSubmitMock).toHaveBeenCalledWith(billingAddress)
})

it("pre-fills the fields if initial billing address is provided", () => {
  const component = renderer.create(<BillingAddress billingAddress={billingAddress} />)

  expect(textInputComponent(component, "Full name").props.value).toEqual("Yuki Stockmeier")
  expect(textInputComponent(component, "Address line 1").props.value).toEqual("401 Broadway")
  expect(textInputComponent(component, "Address line 2 (optional)").props.value).toEqual("25th floor")
  expect(textInputComponent(component, "City").props.value).toEqual("New York")
  expect(textInputComponent(component, "State, Province, or Region").props.value).toEqual("NY")
  expect(textInputComponent(component, "Postal code").props.value).toEqual("10013")
})

const errorTextComponent = (component, label) => findFieldForInput(component, { label }).findByType(Sans12)

const textInputComponent = (component, label) => findFieldForInput(component, { label }).findByType(TextInput)

const findFieldForInput = (component, { label }) => component.root.findByProps({ label })

const billingAddress = {
  fullName: "Yuki Stockmeier",
  addressLine1: "401 Broadway",
  addressLine2: "25th floor",
  city: "New York",
  state: "NY",
  postalCode: "10013",
}
