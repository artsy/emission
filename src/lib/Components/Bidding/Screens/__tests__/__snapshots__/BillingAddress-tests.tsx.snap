// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<View
  className="css-xh2dyr"
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    accessible={true}
    className="css-18ygu8q"
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "testUri": "../../../images/angle-left.png",
      }
    }
    style={
      Array [
        Object {},
        Object {
          "zIndex": 10,
        },
      ]
    }
  />
  <RCTScrollView>
    <View>
      <View
        className="css-3qh1nr-Container"
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          className="css-15qw3qz-Title sc-bwzfXH uTxCW"
          fontFamily="AGaramondPro-Semibold"
          fontSize="22px"
          lineHeight="28px"
          style={
            Array [
              Object {
                "fontFamily": "AGaramondPro-Semibold",
                "fontSize": 22,
                "lineHeight": 28,
              },
            ]
          }
        >
          Your billing address
        </Text>
        <View
          className="css-z2sxea-StyledInput"
          onLayout={[Function]}
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            className="css-1k3s6nb-StyledInput sc-bwzfXH uTxCW"
            fontFamily="AGaramondPro-Regular"
            fontSize="16px"
            lineHeight="24px"
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 16,
                  "lineHeight": 24,
                },
              ]
            }
          >
            Full name
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="words"
            autoFocus={true}
            className="css-15j6ge4-StyledInput"
            error={false}
            inputRef={[Function]}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Add your full name"
            rejectResponderTermination={true}
            returnKeyType="next"
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "height": 40,
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          className="css-z2sxea-StyledInput"
          onLayout={[Function]}
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            className="css-1k3s6nb-StyledInput sc-bwzfXH uTxCW"
            fontFamily="AGaramondPro-Regular"
            fontSize="16px"
            lineHeight="24px"
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 16,
                  "lineHeight": 24,
                },
              ]
            }
          >
            Address line 1
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="words"
            className="css-15j6ge4-StyledInput"
            error={false}
            inputRef={[Function]}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Add your street address"
            rejectResponderTermination={true}
            returnKeyType="next"
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "height": 40,
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          className="css-z2sxea-StyledInput"
          onLayout={[Function]}
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            className="css-1k3s6nb-StyledInput sc-bwzfXH uTxCW"
            fontFamily="AGaramondPro-Regular"
            fontSize="16px"
            lineHeight="24px"
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 16,
                  "lineHeight": 24,
                },
              ]
            }
          >
            Address line 2 (optional)
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="words"
            className="css-15j6ge4-StyledInput"
            error={false}
            inputRef={[Function]}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Add your apt, floor, suite, etc."
            rejectResponderTermination={true}
            returnKeyType="next"
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "height": 40,
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          className="css-z2sxea-StyledInput"
          onLayout={[Function]}
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            className="css-1k3s6nb-StyledInput sc-bwzfXH uTxCW"
            fontFamily="AGaramondPro-Regular"
            fontSize="16px"
            lineHeight="24px"
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 16,
                  "lineHeight": 24,
                },
              ]
            }
          >
            City
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="words"
            className="css-15j6ge4-StyledInput"
            error={false}
            inputRef={[Function]}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Add your city"
            rejectResponderTermination={true}
            returnKeyType="next"
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "height": 40,
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          className="css-z2sxea-StyledInput"
          onLayout={[Function]}
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            className="css-1k3s6nb-StyledInput sc-bwzfXH uTxCW"
            fontFamily="AGaramondPro-Regular"
            fontSize="16px"
            lineHeight="24px"
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 16,
                  "lineHeight": 24,
                },
              ]
            }
          >
            State, Province, or Region
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="words"
            className="css-15j6ge4-StyledInput"
            error={false}
            inputRef={[Function]}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Add state, province, or region"
            rejectResponderTermination={true}
            returnKeyType="next"
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "height": 40,
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          className="css-z2sxea-StyledInput"
          onLayout={[Function]}
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            className="css-1k3s6nb-StyledInput sc-bwzfXH uTxCW"
            fontFamily="AGaramondPro-Regular"
            fontSize="16px"
            lineHeight="24px"
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 16,
                  "lineHeight": 24,
                },
              ]
            }
          >
            Postal code
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="words"
            className="css-15j6ge4-StyledInput"
            error={false}
            inputRef={[Function]}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Add your postal code"
            rejectResponderTermination={true}
            returnKeyType="next"
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "height": 40,
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          className="css-z2sxea-StyledInput"
          onLayout={[Function]}
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            className="css-1k3s6nb-StyledInput sc-bwzfXH uTxCW"
            fontFamily="AGaramondPro-Regular"
            fontSize="16px"
            lineHeight="24px"
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 16,
                  "lineHeight": 24,
                },
              ]
            }
          >
            Phone
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="words"
            className="css-15j6ge4-StyledInput"
            error={false}
            inputRef={[Function]}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Add your phone number"
            rejectResponderTermination={true}
            returnKeyType="next"
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "height": 40,
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          className="css-1itv5e3"
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            className="css-wlea3r sc-bwzfXH uTxCW"
            fontFamily="AGaramondPro-Regular"
            fontSize="16px"
            lineHeight="24px"
            style={
              Array [
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                  "fontSize": 16,
                  "lineHeight": 24,
                },
              ]
            }
          >
            Country
          </Text>
          <View
            accessible={true}
            className="css-xdnqeh"
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              className="css-oi1b5w sc-bwzfXH uTxCW"
              fontFamily="AGaramondPro-Regular"
              fontSize="16px"
              lineHeight="24px"
              style={
                Array [
                  Object {
                    "fontFamily": "AGaramondPro-Regular",
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                ]
              }
            >
              Select your country
            </Text>
          </View>
        </View>
        <View
          height={50}
          style={
            Array [
              Object {
                "height": 50,
              },
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(0, 0, 0, 1)",
                "borderRadius": 2,
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={null}
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 14,
                    },
                    Object {
                      "color": "white",
                      "opacity": 1,
                    },
                    Object {
                      "fontFamily": "Unica77LL-Medium",
                    },
                  ]
                }
              >
                Add billing address
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
